RealmSwift/Realm.swift

```diff
@@ -435,8 +296,10 @@ public class Realm {
 
     :param: object The objects to be deleted.
     */
-    public func delete(objects: Results<Object>) {
-        rlmRealm.deleteObjects(objects)
+    public func delete<S: SequenceType where S.Generator.Element: Object>(objects: S) {
```

Comment:
The Results overload does need to exist. rlmRealm.deleteObjects(objects) does something very different from looping over the items.